src/response/body.rs

```diff
@@ -36,17 +37,26 @@ impl Body {
         let len = body.len();
 
         Self {
-            data: Box::pin(Full::new(body).map_err(Error::from)),
+            data: Box::new(Full::new(body).map_err(Error::from)),
             len: Some(len),
         }
     }
 
     pub(super) fn stream(body: BoxStream) -> Self {
         Self {
-            data: Box::pin(StreamBody::new(body)),
+            data: Box::new(StreamBody::new(body)),
             len: None,
         }
     }
+
+    fn project(self: Pin<&mut Self>) -> Pin<&mut BodyData> {
```

Comment:
A bit of scope creep here but it seemed like a good idea to do a proper pin projection for `data` rather than pinning `data` ahead of time and then using `get_mut().data.as_mut()`.